components/datadog/apps/nginx/eksFargate.go

```diff
@@ -0,0 +1,370 @@
+				Labels: pulumi.StringMap{
+					"app": pulumi.String("nginx"),
+				},
+			},
+			Spec: &policyv1beta1.PodDisruptionBudgetSpecArgs{
+				MaxUnavailable: pulumi.Int(1),
+				Selector: &metav1.LabelSelectorArgs{
+					MatchLabels: pulumi.StringMap{
+						"app": pulumi.String("nginx"),
+					},
+				},
+			},
+		}, opts...); err != nil {
+			return nil, err
+		}
+	} else {
+		if _, err := policyv1.NewPodDisruptionBudget(e.Ctx, "nginx-fargate", &policyv1.PodDisruptionBudgetArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Name:      pulumi.String("nginx"),
+				Namespace: pulumi.String(namespace),
+				Labels: pulumi.StringMap{
+					"app": pulumi.String("nginx"),
+				},
+			},
+			Spec: &policyv1.PodDisruptionBudgetSpecArgs{
+				MaxUnavailable: pulumi.Int(1),
+				Selector: &metav1.LabelSelectorArgs{
+					MatchLabels: pulumi.StringMap{
+						"app": pulumi.String("nginx"),
+					},
+				},
+			},
+		}, opts...); err != nil {
+			return nil, err
+		}
+	}
+
+	if dependsOnCrd != nil {
+		ddm, err := ddv1alpha1.NewDatadogMetric(e.Ctx, "nginx-fargate", &ddv1alpha1.DatadogMetricArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Name:      pulumi.String("nginx"),
+				Namespace: pulumi.String(namespace),
+				Labels: pulumi.StringMap{
+					"app": pulumi.String("nginx"),
+				},
+			},
+			Spec: &ddv1alpha1.DatadogMetricSpecArgs{
+				Query: pulumi.String(fmt.Sprintf("avg:nginx.net.request_per_s{kube_cluster_name:%%%%tag_kube_cluster_name%%%%,kube_namespace:%s,kube_deployment:nginx}.rollup(60)", namespace)),
+			},
+		}, append(opts, dependsOnCrd)...)
+		if err != nil {
+			return nil, err
+		}
+
+		// In versions older than 1.23.0, we should use autoscalingv2beta2
+		kubeThresholdVersion, _ = semver.NewVersion("1.23.0")
+
+		if kubeVersion.Compare(kubeThresholdVersion) < 0 {
+			if _, err := autoscalingv2beta2.NewHorizontalPodAutoscaler(e.Ctx, "nginx-fargate", &autoscalingv2beta2.HorizontalPodAutoscalerArgs{
+				Metadata: &metav1.ObjectMetaArgs{
+					Name:      pulumi.String("nginx"),
+					Namespace: pulumi.String(namespace),
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("nginx"),
+					},
+				},
+				Spec: &autoscalingv2beta2.HorizontalPodAutoscalerSpecArgs{
+					MinReplicas: pulumi.Int(1),
+					MaxReplicas: pulumi.Int(5),
+					ScaleTargetRef: &autoscalingv2beta2.CrossVersionObjectReferenceArgs{
+						ApiVersion: pulumi.String("apps/v1"),
+						Kind:       pulumi.String("Deployment"),
+						Name:       pulumi.String("nginx"),
+					},
+					Metrics: &autoscalingv2beta2.MetricSpecArray{
+						&autoscalingv2beta2.MetricSpecArgs{
+							Type: pulumi.String("External"),
+							External: &autoscalingv2beta2.ExternalMetricSourceArgs{
+								Metric: &autoscalingv2beta2.MetricIdentifierArgs{
+									Name: pulumi.String("datadogmetric@" + namespace + ":nginx"),
+								},
+								Target: &autoscalingv2beta2.MetricTargetArgs{
+									Type:         pulumi.String("AverageValue"),
+									AverageValue: pulumi.String("10"),
```

Comment:
In fact, I recently discovered (See #721) that using `AverageValue` was a mistake because [the query that is done is already computing an average](https://github.com/DataDog/test-infra-definitions/pull/757/files#diff-088146cea769ab392d662d3732d8d047dd3b521f5a78e04d531201cc32ac543fR192):
```
avg:nginx.net.request_per_s{kube_cluster_name:%%%%tag_kube_cluster_name%%%%,kube_namespace:%s,kube_deployment:nginx}.rollup(60)
```

So, as it is today, the total number of TPS received by all Nginx pods it divided twice by the number of pods:
* Once on datadog backend side because of the `avg:` operator in the metrics query.
* Once [by the HPA itself](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/#algorithm-details).

```suggestion
									Type:         pulumi.String("Value"),
									Value: pulumi.String("10"),
```